src/variable.cpp

```diff
@@ -624,8 +643,11 @@ Variable::Variable(const Tag tag, const Unit::Id unit,
                                               std::move(object))) {}
 
 void Variable::setDimensions(const Dimensions &dimensions) {
-  if (dimensions == m_object->dimensions())
+  if (dimensions.volume() == m_object->dimensions().volume()) {
```

Comment:
Does this work as expected? Shouldn't this raise errors if input dimensions are incompatible - or does that happen as part of clone?

Reply:
I agree that this is debatable and I have previously change my mind back and forth about it. It is conceptually similar to the question whether `std::vector` should keep its data when resized (it does).

In this case the array is "multidimensional" so the definition of "keeping" data is not unique. Basically I just went for the definition that requires least computation. Basically this is the least restrictive definition of "input dimensions are compatible" (the underlying 1D array has the correct size to be interpreted as an ND array with given dimensions), therefore not error is thrown.

We could make stricter requirement on compatibility, but then things like `reshape` would need to use a different method to set the dimensions (new method would need to be added to `VariableConcept`).

Reply:
I don't have a strong opinion on this. It might be worth looking at numpy reshape restrictions if you haven't already https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.reshape.html

Reply:
I think you already have in the comment/discussion below.